libr/bin/bfile.c

```diff
@@ -847,3 +848,50 @@ R_API bool r_bin_file_close(RBin *bin, int bd) {
 	}
 	return false;
 }
+
+R_API int r_bin_file_hash_load(RBin *bin, RConfig *config, const char *file) {
+	const ut8 *md5, *sha1;
+	char hash[128], *p;
+	RHash *ctx;
+	ut8* buf;
+	int buf_len = 0;
+	int i;
+	ut64 limit;
+	RBinFile *bf = bin->cur;
+	if (!file && bf) {
+		file = bf->file;
+	}
+	
+	limit = r_config_get_i (config, "cfg.hashlimit");
+	if (bf && bf->size > limit) {
+		return false;
+	}
+	buf = (ut8 *) r_file_slurp (file, &buf_len);
```

Comment:
dangerous, ths can eat all memory.

you shuold use the io api via the RIOBind instance using the fd associated to the RBinFIle selected (dont use bin->cur, take that as argument of the function)